fontforge/python.c

```diff
@@ -2625,14 +2589,172 @@ static PySequenceMethods PyFFContour_Sequence = {
+	return( NULL );
+    }
+#if PY_VERSION_HEX > 0x03060100
+    if (PySlice_Unpack(key, &start, &end, &step) < 0) {
+	return( NULL );
+    }
+    len = PySlice_AdjustIndices(cont->pt_cnt, &start, &end, step);
+#elif PY_VERSION_HEX > 0x03020000
+    if (PySlice_GetIndicesEx(key, cont->pt_cnt, &start, &end, &step, &len) < 0) {
+	return( NULL );
+    }
+#else
+    if (PySlice_GetIndicesEx((PySliceObject *) key, cont->pt_cnt, &start, &end, &step, &len) < 0) {
+	return( NULL );
+    }
+#endif
+    if ( !(step == 1 || step == -1) ) {
+	PyErr_Format(PyExc_IndexError, "Only supported steps are 1 and -1");
+	return( NULL );
+    }
+
+    ret = (PyFF_Contour *)PyFF_ContourType.tp_alloc(&PyFF_ContourType, 0);
+    ret->is_quadratic = cont->is_quadratic;
+    ret->closed = false;
+    ret->pt_max = ret->pt_cnt = len;
+    ret->points = PyMem_New(PyFF_Point *,ret->pt_max);
+
+    for ( i = 0; i < len; i++ ) {
+	ret->points[i] = cont->points[i*step + start];
+	Py_INCREF(ret->points[i]);
+    }
+    return( (PyObject *) ret );
+}
+
+// Caller responsible for incrementing reference count
+static void PyFFContour_CInsertPoint(PyFF_Contour *self, PyFF_Point *p, int pos) {
+	int i;
+
+	if ( pos<0 || pos>=self->pt_cnt-1 )
+		pos = self->pt_cnt-1;
+	if ( self->pt_cnt >= self->pt_max ) {
+		/* Messes with self->points */
+		PyMem_Resize(self->points,PyFF_Point *,self->pt_max += 10);
+	}
+	for ( i=self->pt_cnt-1; i>pos; --i )
+	self->points[i+1] = self->points[i];
+	self->points[pos+1] = p;
+	PyFFContour_ClearSpiros(self);
+	self->pt_cnt += 1;
+}
+
+// Uses a Contour object as a temporary home for the point list
+static PyFF_Contour *PyFFPointList_Parse(PyObject *args) {
+	PyFF_Contour *self = (PyFF_Contour *) PyFFContour_new(&PyFF_ContourType,NULL,NULL);
+	PyFF_Point *p;
+	int len, i;
+	bool ok = true;
+
+	if ( PyType_IsSubtype(&PyFF_ContourType, Py_TYPE(args) ) ) {
+		PyFFContour_InPlaceConcat((PyObject *) self, args);
+	} else if ( PySequence_Check(args) ) {
+		len = PySequence_Size(args);
+		for (i = 0; i < len; ++i) {
+			p = PyFFPoint_Parse(PySequence_GetItem(args, i), false, false);
+			if ( p!=NULL ) {
+				PyFFContour_CInsertPoint(self, p, -1);
+			} else {
+				ok = false;
+				break;
+			}
+		}
+	} else {
+		ok = false;
+	}
+	if ( ! ok ) {
+		PyFFContour_dealloc(self);
+		return NULL;
+	}
+	return self;
+}
+
+static int PyFFContour_SubAssign( PyObject *self, PyObject *key, PyObject *val ) {
+    PyFF_Contour *cont = (PyFF_Contour *) self, *rpl;
+    int i, diff;
+    Py_ssize_t len, start, end, step;
+
+    if ( PyInt_Check(key)) {
+        return PyFFContour_IndexAssign(self, PyNumber_AsSsize_t(key, PyExc_IndexError), val);
+    }
+    rpl = PyFFPointList_Parse(val);
+    if ( rpl==NULL ) { 
+	PyErr_Format(PyExc_TypeError, "Replacement must encode a point list");
+	return( -1 );
+    }
+    if ( !PySlice_Check(key) ) { 
+	PyErr_Format(PyExc_IndexError, "Contour indexed by integer only");
+	return( -1 );
+    }
+#if PY_VERSION_HEX > 0x03060100
```

Comment:
It feels like this slice unpacking code should be either macro'ed or made into a function

Reply:
It's only two instances ...